src/flypad-backend/src/Aircraft/AircraftProcedures.h

```diff
@@ -16,285 +19,284 @@ struct ProcedureStep {
+  static constexpr std::size_t COLD_DARK_SIZE = TAKEOFF_CONFIG_OFF.size() + TAXI_CONFIG_OFF.size() + PUSHBACK_CONFIG_OFF.size() + POWERED_CONFIG_OFF.size();
+  static constexpr std::size_t POWERED_SIZE = TAKEOFF_CONFIG_OFF.size() + TAXI_CONFIG_OFF.size() + PUSHBACK_CONFIG_OFF.size() + POWERED_CONFIG_ON.size();
+  static constexpr std::size_t RDY_FOR_PUSHBACK_SIZE = TAKEOFF_CONFIG_OFF.size() + TAXI_CONFIG_OFF.size() + POWERED_CONFIG_ON.size() + PUSHBACK_CONFIG_ON.size();
+  static constexpr std::size_t RDY_FOR_TAXI_SIZE = TAKEOFF_CONFIG_OFF.size() + POWERED_CONFIG_ON.size() + PUSHBACK_CONFIG_ON.size() + TAXI_CONFIG_ON.size();
+  static constexpr std::size_t RDY_FOR_TAKEOFF_SIZE = POWERED_CONFIG_ON.size() + PUSHBACK_CONFIG_ON.size() + TAXI_CONFIG_ON.size() + TAKEOFF_CONFIG_ON.size();
+
+  std::array<const ProcedureStep*, COLD_DARK_SIZE> coldAndDark;
+  std::array<const ProcedureStep*, POWERED_SIZE> powered;
+  std::array<const ProcedureStep*, RDY_FOR_PUSHBACK_SIZE> readyForPushback;
+  std::array<const ProcedureStep*, RDY_FOR_TAXI_SIZE> readyForTaxi;
+  std::array<const ProcedureStep*, RDY_FOR_TAKEOFF_SIZE> readyForTakeoff;
+
+  template<std::size_t N1, std::size_t N2>
+  static constexpr void doInsertProcedures(std::array<const ProcedureStep*, N1>& dest, const std::array<ProcedureStep, N2>& src, const std::size_t offset) {
+    std::transform(begin(src), end(src), begin(dest) + offset, [](const auto& procedure) {
+      return &procedure;
     });
-#endif
-
-    coldAndDark->insert(coldAndDark->end(), TAKEOFF_CONFIG_OFF->begin(), TAKEOFF_CONFIG_OFF->end());
-    coldAndDark->insert(coldAndDark->end(), TAXI_CONFIG_OFF->begin(), TAXI_CONFIG_OFF->end());
-    coldAndDark->insert(coldAndDark->end(), PUSHBACK_CONFIG_OFF->begin(), PUSHBACK_CONFIG_OFF->end());
-    coldAndDark->insert(coldAndDark->end(), POWERED_CONFIG_OFF->begin(), POWERED_CONFIG_OFF->end());
+  }
 
-    powered->insert(powered->end(), TAKEOFF_CONFIG_OFF->begin(), TAKEOFF_CONFIG_OFF->end());
-    powered->insert(powered->end(), TAXI_CONFIG_OFF->begin(), TAXI_CONFIG_OFF->end());
-    powered->insert(powered->end(), PUSHBACK_CONFIG_OFF->begin(), PUSHBACK_CONFIG_OFF->end());
-    powered->insert(powered->end(), POWERED_CONFIG_ON->begin(), POWERED_CONFIG_ON->end());
+  template<std::size_t... I>
+  static constexpr void doInsertProcedures(auto& dest, const auto& sources, const auto& offsets, std::index_sequence<I...>) {
+    (doInsertProcedures(dest, std::get<I>(sources), offsets[I]), ...);
+  }
 
-    readyForPushback->insert(readyForPushback->end(), TAKEOFF_CONFIG_OFF->begin(), TAKEOFF_CONFIG_OFF->end());
-    readyForPushback->insert(readyForPushback->end(), TAXI_CONFIG_OFF->begin(), TAXI_CONFIG_OFF->end());
-    readyForPushback->insert(readyForPushback->end(), POWERED_CONFIG_ON->begin(), POWERED_CONFIG_ON->end());
-    readyForPushback->insert(readyForPushback->end(), PUSHBACK_CONFIG_ON->begin(), PUSHBACK_CONFIG_ON->end());
+  template<class Tuple, std::size_t... I>
```

Comment:
Please revert these - these are much more than simple improvements. Especially these templates are much less readable. Not everything needs to be complicated sophisticated cpp code for no good reason.